apps/system/utils/utils_cpuload.c

```diff
@@ -116,18 +117,30 @@ static void cpuload_stop(void)
  * So for zero load case, we check if avg_load is 0.0 or not.
  * Note: we compare with "-0.0" because if avg_load is 90.0, then
  * also we will get true if we compare with 0.0 */
+
 static bool has_cpuload(char *load_info)
 {
 #ifdef CONFIG_SMP
 	int len = strlen(load_info);
-	if (strcmp(&load_info[len - 4], "-0.0") == 0) {
-#else
-	if (strcmp(load_info, "0.0") == 0) {
-#endif
-		return false;
-	}
 
-	return true;
+	if (cpu == CONFIG_SMP_NCPUS) {
+		/* This is the case when user has not specified cpu idx. 
+		 * In this case we check the avg value. If avg is zero, then
+		 * all cpu load values will be zero and hence we return false.
+		 */
+		return (strncmp(&load_info[len - 4], "-0.0", 4) != 0);
+	} else if (cpu == 0) {
+		return (strncmp(load_info, "0.0", 3) != 0);
+	} else {
+		char *str = load_info;
+		for (int i = 1; i < CONFIG_SMP_NCPUS && i <= cpu; i++) {
+			str = strchr(str, '-') + 1;
+		}
+		return (strncmp(str, "0.0", 3) != 0);
+	}
+#else 
+	return (strcmp(load_info, "0.0") != 0);
```

Comment:
If you don't use the `strncmp`, SVACE will give you a warning I think. And other lines already use strncmp.

And Is it ok with length 3 for "0.0"? not 4? ok with length 4 for "-0.0"?

Reply:
Changed to strncmp. 
And it is ok to compare just 3 chars for 0.0 and 4 for -0.0 because it might not be end of string.